src/routes/users.ts

```diff
@@ -0,0 +1,74 @@
+import {Router, Request, Response} from "express";
+import {getAutoSuggestUsers} from "../services/user.service";
+import {userQuerySchema, UserRequestBodySchema} from "../validationSchemes/schemes";
+import {ValidatedRequest, createValidator, ExpressJoiInstance} from "express-joi-validation";
+
+const router = Router();
+const mongoose = require("mongoose");
+const validator: ExpressJoiInstance = createValidator({
+  passError: true
+});
+const User = mongoose.model("User", {
+  id: String,
+  login: String,
+  password: String,
+  age: Number,
+  isDeleted: Boolean,
+});
+const USERS_LIMIT: number = 5;
+const loginSubstring: string = 'ill';
+
+router.get("/users", (request: Request, response: Response) => {
+  User.find({}, (error, users) => {
+    if(error) return console.log(error);
+    const filteredUsers = getAutoSuggestUsers(users, loginSubstring, USERS_LIMIT);
+
+    response.send(filteredUsers);
+  });
+});
+
+router.get("/users/:id", (request, response) => {
+  User.find({ id: request.params.id }, (error, message) => {
+    if(error) return console.log(error);
+    
+    response.send(message);
+  });
+});
+
+router.post("/users", validator.body(userQuerySchema), (request: ValidatedRequest<UserRequestBodySchema>, response: Response) => {
+    if(!request.body) return response.sendStatus(400);

```

Comment:
You've already added validation `validator.body(userQuerySchema)`, is that row extra?